src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -222,10 +226,16 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                 return false;
             }
 
-            // Set up the DX Engine
-            auto dxEngine = std::make_unique<::Microsoft::Console::Render::DxEngine>();
-            _renderer->AddRenderEngine(dxEngine.get());
-            _renderEngine = std::move(dxEngine);
+            if (Feature_AtlasEngine::IsEnabled() && _settings.UseAtlasEngine())
+            {
+                _renderEngine = std::make_unique<::Microsoft::Console::Render::AtlasEngine>();
+            }
+            else
+            {
+                _renderEngine = std::make_unique<::Microsoft::Console::Render::DxEngine>();
+            }

```

Comment:
Shorter code isn't really a meaningful goal on its own :)

Reply:
I do like the ternary, but it doesn't make the code simpler unfortunately...
Ternary operators require both sides to be of the same type. You'd have to write:
```cpp
_renderEngine  = Feature_AtlasEngine::IsEnabled() && _settings.UseAtlasEngine() ?
    static_cast<std::unique_ptr<IRenderEngine>>(std::make_unique<::Microsoft::Console::Render::AtlasEngine>()) :
    static_cast<std::unique_ptr<IRenderEngine>>(std::make_unique<::Microsoft::Console::Render::DxEngine>());
```
...which is hard to read.

The second variant doesn't work because template parameters must be `constexpr` and `_settings.UseAtlasEngine()` isn't.